apps/advisor/pkg/app/checkscheduler/checkscheduler.go

```diff
@@ -133,19 +136,40 @@ func (r *Runner) checkLastCreated(ctx context.Context) (time.Time, error) {
 		if itemCreated.After(lastCreated) {
 			lastCreated = itemCreated
 		}
+
+		// If the check is unprocessed, set it to error
+		if checks.GetStatusAnnotation(item) == "" {
+			log.Error("Check is unprocessed", "check", item.GetStaticMetadata().Identifier())
+			err := checks.SetStatusAnnotation(ctx, r.client, item, checks.StatusAnnotationError)
+			if err != nil {
+				log.Error("Error setting check status to error", "error", err)
+			}
+		}
 	}
 	return lastCreated, nil
 }
 
 // createChecks creates a new check for each check type in the registry.
 func (r *Runner) createChecks(ctx context.Context) error {
-	for _, check := range r.checkRegistry.Checks() {
+	// List existing CheckType objects
+	list, err := r.typesClient.List(ctx, r.namespace, resource.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("error listing check types: %w", err)
+	}
+
+	// Create checks for each CheckType
+	for _, item := range list.GetItems() {
+		checkType, ok := item.(*advisorv0alpha1.CheckType)
+		if !ok {
+			continue
+		}
```

Comment:
This fixes an issue that sometimes, not all the check types are properly saved. Since `r.checkRegistry.Checks` returns the hardcoded list of check types, this was creating checks for types that were not properly registered, leading to errors like this:

```
msg="Error processing check" app=advisor.app check=ssosetting error="checktypes.advisor.grafana.app \"ssosetting\" not found"
```

Rather than that, we are now listing the check types dynamically, using the API.